vulture/core.py

```diff
@@ -601,6 +612,18 @@ def visit(self, node):
             self._log(lineno, ast.dump(node), line)
         if visitor:
             visitor(node)
+
+        # There isn't a clean subset of node types that might have type
+        # comments, so just check all of them.
+        type_comment = getattr(node, "type_comment", None)
+        if type_comment is not None:
+            mode = (
+                "func_type"
+                if isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef))
+                else "eval"
+            )
+            self.visit(ast.parse(type_comment, "<type_comment>", mode))
```

Comment:
Let's use keys to make the call clearer: `ast.parse(type_comment, filename="<type_comment>", mode=mode)`